### DISCUSSION TITLE
Prisma suddenly throwing PrismaClientKnownRequestError without any changes to the database or the client

### DISCUSSION THREAD

**Discussion by manuelpoelzl** [prisma/prisma#22333] [Q&A] [answered]
### Question

Today prisma started to throw a `PrismaClientKnownRequestError` without a valid reason in my opionion.
The problematic  model looks like this:
```prisma
model ArtifactImage {
  ID         String   @id @unique @db.VarChar(36)
  Artifact   Artifact @relation(fields: [ArtifactId], references: [ID], onDelete: Cascade)
  ArtifactId Int      @map("Artifact")
  Extension  String   @db.VarChar(5)

  @@map("artifact_images")
}
```
I'm trying to perform a createMany operation like this:
![Screenshot_17](https://github.com/prisma/prisma/assets/46539758/66092176-c848-4457-a28a-523faf3afd17)

My foreign key constraints are met, and each column has the right type and length.
The result of invoking the above mentioned query is this:
![Screenshot_15](https://github.com/prisma/prisma/assets/46539758/8da6223a-ee26-46bf-a527-4cf365dc3fe1)

If this is thrown, i do not get in the catch handler this function is wrapped with, the whole App just crashes.
And whats even funnier - the request fails, but the data is inserted successfully in the database:
![image](https://github.com/prisma/prisma/assets/46539758/fc695012-6755-4385-ad60-686cd245589e)

So I'm kinda confused why this happens now, the only thing that changed recently, was that i removed XAMPP for my testing and switched to MariaDB server, but I think this should not be a problem since all other queries except this one work perfectly fine.

I am grateful for any help!

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows 10 Pro
- Database: MariaDB
- Node.js version: 20.10.0

prisma -v output:

Environment variables loaded from .env
prisma                  : 5.6.0                                                                                                                   
@prisma/client          : 5.6.0                                                                                                                   
Computed binaryTarget   : windows                                                                                                                 
Operating System        : win32                                                                                                                   
Architecture            : x64                                                                                                                     
Node.js                 : v20.10.0                                                                                                                
Query Engine (Node-API) : libquery-engine e95e739751f42d8ca026f6b910f5a2dc5adeaeee (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli e95e739751f42d8ca026f6b910f5a2dc5adeaeee (at node_modules\@prisma\engines\schema-engine-windows.exe)  
Schema Wasm             : @prisma/prisma-schema-wasm 5.6.0-32.e95e739751f42d8ca026f6b910f5a2dc5adeaeee                                            
Default Engines Hash    : e95e739751f42d8ca026f6b910f5a2dc5adeaeee                                                                                
Studio                  : 0.495.0

**Comment by ludralph** [comment, marked as answer]
Hi @manuelpoelzl 👋 

Thank you for raising this question.

Can you please share the complete schema file and the query you a performing? Also, can you share the complete error log as well? You can also enable [`debugging`](https://www.prisma.io/docs/orm/prisma-client/debugging-and-troubleshooting/debugging#setting-the-debug-environment-variable) so that we can have access to the detailed error trace.

**Reply by manuelpoelzl** [reply]
I have found the cause of the problem!
The error is correct, but it is a consequential error.

An attempt is made to delete a data record that does not exist, which of course then fails. This error in turn is tried to be written in the logs table of the database and here the error mentioned above actually applies, because the message to be written is longer than the length value of my varchar in the table.

A merge seems to have caused problems.
Thanks to your linked page on how to enable debugging, I was able to find the error

Thank you!
